### PR TITLE
Implementation of flood fill.

### PR DESCRIPTION
**Pull request by algodiva**
Generic implementation of span based FFA. Should work for any grid type that can be delineated by an x and y coordinate system, supports custom comparison functors for fine-tuned control of filling. Works for any data type. Any comparisons are done in the functor, and the user will have full control of what comparison operations are allowed.

### CODE REVIEW

**Comment by algodiva** [comment]
Yea this coverage requirement for the display trait is not correct. That should never be blocking this from going through.

File: src/core/algorithms.rs [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+use ndarray::Array2;
+
+/// Error for flood_fill
+#[derive(Debug)]
+pub enum FFError {
+    /// Denotes an invalid seed point was provided.
+    InvalidSeed,
+}
+
+impl Display for FFError {
```
Comment (sevriarch) [inline src/core/algorithms.rs:14]: Do we know why (according to codecov) this doesn't get exercised during the test that generates an error?

Comment (algodiva) [reply src/core/algorithms.rs:14]: Because the value is not printed, the error is just thrown. I'm trying to fix it so codecov behaves.

Comment (sevriarch) [reply src/core/algorithms.rs:14]: Gotcha. I assume we can do something like test the error message?

**Review by sevriarch — APPROVED** [review]
LGTM

**Pull request merged**